venafi/resource_venafi_certificate.go

```diff
@@ -361,9 +362,63 @@ func resourceVenafiCertificateUpdate(_ context.Context, d *schema.ResourceData,
 	return nil
 }
 
-func resourceVenafiCertificateDelete(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
-	// We don't support deletion for created certificates from TPP, so we just remove it from state
+func resourceVenafiCertificateDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+
+	//casting meta interface to the expected *venafiProviderConfig
+	provConf, ok := meta.(*venafiProviderConfig)
+	if !ok {
+		castError := errors.New(messageVenafiProviderConfigCastingFailed)
+		tflog.Error(ctx, castError.Error())
+		return diag.FromErr(castError)
+	}
+	vCertCfg := provConf.vCertCfg
+
+	//VCert connector fake has not implemented the RetireCertificate functionality
+	if provConf.skipRetirement || vCertCfg.ConnectorType == endpoint.ConnectorTypeFake {
+		// removing it from state
+		d.SetId("")
+		return nil
+	}
+
+	cl, err := getConnection(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	certID := d.Id()
+	parameters := strings.Split(certID, ",")
+
+	var pickupID string
```

Comment:
you should put this lines after the conditional for parameters, that way you have guaranteed to always have a value at index 0 and can reduce the code from:

```
var pickupID string
pickupID = parameters[0]
```
to
```
pickupID := parameters[0]
```

Reply:
done.